blueman/plugins/applet/ConnectionNotifier.py

```diff
@@ -13,10 +16,23 @@ class ConnectionNotifier(AppletPlugin):
     __description__ = _("Shows desktop notifications when devices get connected or disconnected.")
 
     def on_device_property_changed(self, path: str, key: str, value: Any) -> None:
-        if key == "Connected":
-            device = Device(obj_path=path)
-            if value and "Battery" in self.parent.Plugins.get_loaded() and Battery.applicable(device):
-                return
+        def show_notification() -> bool:
+            try:
+                perc = f"{battery['Percentage']}%"
+                icon_name = "battery"
+            except BluezDBusException:
+                perc = None
+                icon_name = device["Icon"]
+                logging.debug("Failed to get battery level")
 
-            Notification(device["Alias"], _("Connected") if value else _("Disconnected"),
-                         icon_name=device["Icon"]).show()
+            txt = f"{_('Connected')} {perc if perc is not None else ''}"
+            Notification(device["Alias"], txt, icon_name=icon_name).show()
+            return False
+
+        device = Device(obj_path=path)
+        battery = Battery(obj_path=path)
+
+        if key == "Connected" and value:
+            GLib.timeout_add_seconds(5, show_notification)
```

Comment:
I really don't like the generic delay but let's get this merged an I'll test a little bit with notification replacements on top of it (and for other topics).

Reply:
The notification stuff itself is easy. `_NotificationBubble` actually already handles replacing. See #1736. On top of that we just need to keep track of the notification (with a device -> notification mapping I guess).